xcm/xcm-simulator/src/lib.rs

```diff
@@ -161,13 +168,26 @@ macro_rules! __impl_ext {
 				$ext_name.with(|v| *v.borrow_mut() = $new_ext);
 			}
 
-			fn execute_with<R>(execute: impl FnOnce() -> R) -> R {
+			fn execute_without_auto_dispatch<R>(execute: impl FnOnce() -> R) -> R {
 				$ext_name.with(|v| v.borrow_mut().execute_with(execute))
 			}
+
+			fn execute_with<R>(execute: impl FnOnce() -> R) -> R {
+				let result = $ext_name.with(|v| v.borrow_mut().execute_with(execute));
+				process_messages().expect("message processing failure");
```

Comment:
Including returned `XcmResult` in `expect` str would be helpful to give more details on panic.